core/src/library.rs

```diff
@@ -564,8 +564,13 @@ impl<'gc> Library<'gc> {
                 data,
             } => {
                 let descriptor = FontDescriptor::from_parts(&name, is_bold, is_italic);
-                if let Ok(font) = Font::from_font_file(gc_context, descriptor, Cow::Owned(data), 0)
-                {
+                if let Ok(font) = Font::from_font_file(
+                    gc_context,
+                    descriptor,
+                    Cow::Owned(data),
+                    0,
+                    FontType::Device,
+                ) {
                     let name = font.descriptor().name().to_owned();
                     info!("Loaded new device font \"{name}\" from file");
```

Comment:
This message is triggering but it's not actually a device font

Reply:
It is? We are inside `register_device_font`.

Reply:
But DefineFont4 is embedding a font

Reply:
Correct and for DefineFont4 we use `register_character`.

Reply:
Spooky. Loading this branch and the test I sent you - but deleting the Exo2 font from my computer - I see this in Ruffle's log:

```
2024-01-19T17:58:56.800120Z  INFO run_frame: ruffle_desktop::backends::ui: Loading device font "Exo2-Regular" for "Exo 2" (italic: false, bold: false)
2024-01-19T17:58:56.800415Z  INFO run_frame: ruffle_core::library: Loaded new device font "Exo 2" from file
```

It shouldn't be able to find that font (it physically doesn't exist) and it shouldn't even be looking for that font in the first place (it's embedded)

Reply:
We might be looking up the font as a device font, because `display_object.set_is_device_font` happens after `display_object.set_text_format`, so first relayout and assume the font is a device font?

Reply:
Makes sense!

Reply:
Yep. Reordering fixed this.